chapters/my_first_jina_app/index.md

```diff
@@ -0,0 +1,389 @@
+
+Just as a plant manages nutrient flow and growth rate for its branches, Jina's Flow manages the states and context of a group of Pods, orchestrating them to accomplish one specific task. 
+
+We define Flows in `app.py` to index and query our dataset:
+
+```python
+from jina.flow import Flow
+
+<other code here>
+
+def index():
+    f = Flow.load_config('flows/index.yml')
+
+    with f:
+        data_path = os.path.join(os.path.dirname(__file__), os.environ.get('JINA_DATA_FILE', None)) # Set data path
+        f.index_lines(filepath=data_path, batch_size=16, read_mode='r', size=num_docs) # Set mode (index_lines) and indexing settings
+```
+
+To start the Flow, we run `python app.py -t <flow_name>`, in this case:
+
+```sh
+python app.py -t index
+```
+
+ℹ️  `-t` is short for `--task`
+ℹ️  You also can build Flows in `app.py` itself [without specifying them in YAML](https://docs.jina.ai/chapters/flow/index.html) or with [Jina Dashboard](http://dashboard.jina.ai)
+
+#### Indexing
+
+`input.txt` is just one big text file. Our indexing Flow will create an index of each line in the file, and later Jina will query this index. The indexing is performed by the Pods in the Flow. Each Pod performs a different task, with one Pod's output becoming another Pod's input. 
+
+Every Flow is defined in its own YAML file. Let's look at `flows/index.yml`:
+
+```yaml
+!Flow
+version: '1'
+pods:
+  - name: encoder
+    uses: pods/encode.yml
+    timeout_ready: 1200000
+    read_only: true
+  - name: indexer
+    uses: pods/index.yml
+```
+
+Each Pod performs a different operation on the dataset:
+
+| Pod       | Task                                                       |
+| ---       | ---                                                        |
+| `crafter` | Split Documents into sentences                             |
+| `encoder` | Encode each input Document into a vector                   |
+| `indexer` | Build an index of the vectors and metadata key-value pairs |
+
+#### Searching
+
+Like the index Flow, the search Flow is also defined in a YAML file, in this case at `flows/query.yml`:
+
+```yaml
+!Flow
+version: '1'
+with:
+  read_only: true
+  port_expose: $JINA_PORT
+pods:
+  - name: encoder
+    uses: pods/encode.yml
+    timeout_ready: 60000
+    read_only: true
+  - name: indexer
+    uses: pods/index.yml
+```
+
+As in `flows/index.yml`, we use three Pods, but this time they behave differently:
+
+| Pod       | Task                                                              |
+| ---       | ---                                                               |
+| `crafter` | Split input query into sentences                                  |
+| `encoder` | Encode user's query into a vector                                 |
+| `indexer` | Query vector index and key-value pairs; Return matching Documents |
+
+### Pods
+
+<img src=""https://docs.jina.ai/_images/pod.png width="20%" align="left">
```

Comment:
This image is not showing.
```suggestion
<img src="https://docs.jina.ai/_images/pod.png width="20%" align="left">
```

Reply:
For ToC is there an automatic way to do this @bwanglzu @FionnD ? Otherwise it's a major time-sink to create and (more importantly) maintain

Reply:
Fixed image

Reply:
https://ecotrust-canada.github.io/markdown-toc/